reconcile/requests_sender.py

```diff
@@ -69,33 +71,35 @@ def run(dry_run):
     credentials_requests_to_send = \
         [r for r in credentials_requests if not state.exists(r['name'])]
     for credentials_request_to_send in credentials_requests_to_send:
-        user = credentials_request_to_send['user']
-        org_username = user['org_username']
-        public_gpg_key = user.get('public_gpg_key')
-        credentials_name = credentials_request_to_send['credentials']
-        if not public_gpg_key:
+        try:
+            user = credentials_request_to_send['user']
+            credentials_name = credentials_request_to_send['credentials']
+            org_username = user['org_username']
+            logging.info(['send_credentials', org_username, credentials_name])
+
+            if not dry_run:
+                request_name = credentials_request_to_send['name']
+                names = [org_username]
+                subject = request_name
+                encrypted_credentials = get_encrypted_credentials(
+                    credentials_name, user, settings, smtp_client
+                )
+                body = MESSAGE_TEMPLATE.format(
+                    request_name, credentials_name, encrypted_credentials
+                )
+                smtp_client.send_mail(names, subject, body)
+                state.add(request_name)
+        except KeyError:
```

Comment:
i understand where you are coming from here. fwiw, we have schema validation that verifies all our assumptions, and only `public_gpg_key` is not mandatory. it's only a matter of taste, but i personally prefer short(er?) try blocks, and only where an issue may occur. leaving it for your judgement, as this is perfectly fine the way it is.

Reply:
Yeah, but if I capture the exception here I can avoid an ugly if that breaks the flow of this block. This `try` is on the long end, also on the limit of what I find acceptable, TBH.